src/compiler/checker.ts

```diff
@@ -20857,7 +20857,7 @@ namespace ts {
         // flags for the string, number, boolean, "", 0, false, void, undefined, or null types respectively. Returns
         // no flags for all other types (including non-falsy literal types).
         function getFalsyFlags(type: Type): TypeFlags {
-            return type.flags & TypeFlags.Union ? getFalsyFlagsOfTypes((type as UnionType).types) :
+            return type.flags & TypeFlags.UnionOrIntersection ? getFalsyFlagsOfTypes((type as UnionType).types) :

```

Comment:
I'm a little surprised that adding intersections to getFalsyFlags didn't change any existing baselines.

Reply:
Well, there are precious few intersections that would be affected, pretty much only those like `T & undefined` that will now appear to be falsy.